samples/nrf9160/modem_shell/src/gnss/gnss_interface.c

```diff
@@ -617,20 +856,25 @@ static void gnss_api_init(void)
 	if (!err) {
 		gnss_api_initialized = true;
 
+		struct k_work_queue_config gnss_work_q_config = {
+			.name = "gnss_workq",
+			.no_yield = false
+		};
+
 		k_work_queue_start(
 			&gnss_work_q,
 			gnss_workq_stack_area,
 			K_THREAD_STACK_SIZEOF(gnss_workq_stack_area),
 			GNSS_WORKQ_THREAD_PRIORITY,
-			NULL);
+			&gnss_work_q_config);
 
 		k_work_init(&gnss_stop_work, stop_gnss_work_fn);
 		k_work_init_delayable(&gnss_start_work, start_gnss_work_fn);
 		k_work_init_delayable(&gnss_timeout_work, handle_timeout_work_fn);
 
-#if defined(CONFIG_SUPL_CLIENT_LIB)
+#if defined(CONFIG_NRF_CLOUD_AGPS) || defined(CONFIG_SUPL_CLIENT_LIB)
 		k_work_init(&get_agps_data_work, get_agps_data);
-
+#if !defined(CONFIG_NRF_CLOUD_AGPS)
```

Comment:
I think this could be `#if defined(CONFIG_SUPL_CLIENT_LIB)`, right?
And then this doesn't need to be within `#if defined(CONFIG_NRF_CLOUD_AGPS) || defined(CONFIG_SUPL_CLIENT_LIB)` but its `#endif` can be before this line.

Reply:
I guessed that somebody will ask about this one. :) This is `!defined(CONFIG_NRF_CLOUD_AGPS)` in case somebody enables both nRF Cloud A-GPS and SUPL at the same time. In that case nRF Cloud A-GPS would get selected and this prevents unnecessary SUPL initialization.

There's already a build assert for not allowing P-GPS and SUPL at the same time. I could change that so that neither nRF Cloud A-GPS nor P-GPS is allowed with SUPL. Then I could change this one to `defined(CONFIG_SUPL_CLIENT_LIB)`. I'll do that.